From f226774ae4c82210eaaa99f1093c0977de61bbce Mon Sep 17 00:00:00 2001
From: "Brian J. Tarricone" <brian@tarricone.org>
Date: Thu, 3 Aug 2023 12:50:11 -0700
Subject: [PATCH] Ensure all various XDG_* etc. env vars are passed to DBus and
 systemd

Services (such as tumblerd and xfce4-notifyd) are started either by DBus
or by systemd, and need to know things such as XDG_CONFIG_HOME and
XDG_CACHE_HOME.

Closes #177

(cherry picked from commit 0fe26b4d0eab7207c5349af340eb9e8cb833e29c)
---
 scripts/xinitrc.in.in | 31 +++++++++++++++++++++++++++----
 1 file changed, 27 insertions(+), 4 deletions(-)

diff --git a/scripts/xinitrc.in.in b/scripts/xinitrc.in.in
index 9bd6fc3c..da82e079 100755
--- a/scripts/xinitrc.in.in
+++ b/scripts/xinitrc.in.in
@@ -80,11 +80,34 @@ cat /dev/null $XRESOURCES | xrdb -merge -
 # load local modmap
 test -r $HOME/.Xmodmap && xmodmap $HOME/.Xmodmap
 
-# if XAUTHLOCALHOSTNAME is not set in systemd user session, starting of xfce4-notifyd, DISPLAY etc. will fail
-if command -v systemctl >/dev/null 2>&1 && systemctl --user list-jobs >/dev/null 2>&1; then # user session is running
-  dbus-update-activation-environment --systemd XAUTHLOCALHOSTNAME=$XAUTHLOCALHOSTNAME
-fi
+# ensure both the DBus session bus and systemd user session (if running) has
+# all the env vars it needs to properly populate the environment of child
+# processes
+if command -v dbus-update-activation-environment >/dev/null 2>&1; then
+  if command -v systemctl >/dev/null 2>&1 && systemctl --user list-jobs >/dev/null 2>&1; then # user session is running
+    systemd_arg='--systemd'
+  fi
 
+  dbus-update-activation-environment $systemd_arg \
+    DESKTOP_SESSION \
+    XAUTHLOCALHOSTNAME=$XAUTHLOCALHOSTNAME \
+    XDG_CACHE_HOME \
+    XDG_CONFIG_DIRS \
+    XDG_CONFIG_HOME \
+    XDG_CURRENT_DESKTOP \
+    XDG_DATA_DIRS \
+    XDG_DATA_HOME \
+    XDG_MENU_PREFIX \
+    XDG_RUNTIME_DIR \
+    XDG_SEAT \
+    XDG_SEAT_PATH \
+    XDG_SESSION_CLASS \
+    XDG_SESSION_DESKTOP \
+    XDG_SESSION_ID \
+    XDG_SESSION_PATH \
+    XDG_SESSION_TYPE \
+    XDG_STATE_HOME
+fi
 
 # check if we start xfce4-session with ck-launch-session. this is only
 # required for starting from a console, not a login manager
-- 
GitLab

